### PR TITLE
Add build release script

### PR DESCRIPTION
**Pull request by yussufsh**
Signed-off-by: Yussuf Shaikh <yussuf@us.ibm.com>

This script can be used to build the provider plug-in for multiple platforms.

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+# If required for only current platform set LOCAL=1
+if [ "${LOCAL}x" != "x" ]; then
+    XC_OS=$(go env GOOS)
+    XC_ARCH=$(go env GOARCH)
+fi
+
+if ! which gox > /dev/null; then
+    echo "==> Installing gox..."
+    go get -u github.com/mitchellh/gox
```
Comment (LorbusChris) [inline scripts/build.sh]: this will add `gox` (and `github.com/hashicorp/go-version`) to `go.mod`. Could you add `gox` to `tools.go` instead?

**Review by LorbusChris — COMMENTED** [review]
Thank you for the contribution!
Maybe you could also add a `make binaries` command (or similar) for invoking this script to the Makefile?

**Comment by yussufsh** [comment]
> 
> 
> Thank you for the contribution!
> Maybe you could also add a `make binaries` command (or similar) for invoking this script to the Makefile?

`make release` it is!

File: scripts/build.sh (hunk 1 above)
Comment (LorbusChris) [reply scripts/build.sh]: I think now we just need `go install` here now to use the locally vendored version and prevent this script from making changes to go.mod

Comment (LorbusChris) [reply scripts/build.sh]: also please run `go mod tidy && go mod vendor` and include the changes in this commit.

Comment (yussufsh) [reply scripts/build.sh]: @LorbusChris Sorry for the delay.

I have made it `go install` and also `go mod tidy && go mod vendor` did not have any change.

Comment (yussufsh) [reply scripts/build.sh]: Ah you meant to add `go mod tidy && go mod vendor` in the script.. done!

**Review by LorbusChris — APPROVED** [review]

**Pull request merged**